src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -184,6 +185,27 @@ public String toJSON(MaxwellOutputConfig outputConfig) throws Exception {
 			g.writeObjectField(entry.getKey(), entry.getValue());
 		}
 
+		EncryptionContext encryptionContext = null;
+		if (outputConfig.encryptionEnabled()) {
+			encryptionContext = EncryptionContext.create(outputConfig.secretKey);
+		}
+
+		DataJsonGenerator dataWriter = outputConfig.encryptionMode == EncryptionMode.ENCRYPT_DATA
+			? json.getEncryptingGenerator()
+			: json.getPlaintextGenerator();
+
+		JsonGenerator dataGenerator = dataWriter.begin();
```

Comment:
this diff is weird, why did all this code move above there?

Reply:
We are using the dataWriter to write the primaryKeys and primaryKeyColumn objects but we need to write them before the excludeColumns logic happens. If we don't, any of the columns that make up the primary key that are also in the exlcudeColumns list will not be outputted.